src/main/java/org/broadinstitute/hellbender/tools/exome/orientationbiasvariantfilter/OrientationBiasFilterer.java

```diff
@@ -0,0 +1,361 @@
+package org.broadinstitute.hellbender.tools.exome.orientationbiasvariantfilter;
+
+import com.google.cloud.dataflow.sdk.repackaged.com.google.common.annotations.VisibleForTesting;
+import htsjdk.variant.variantcontext.*;
+import htsjdk.variant.vcf.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.engine.ProgressMeter;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.genotyper.IndexedSampleList;
+import org.broadinstitute.hellbender.utils.genotyper.SampleList;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Perform operations for the orientation bias filter.
+ */
+public class OrientationBiasFilterer {
+
+    private static final Logger logger = LogManager.getLogger(OrientationBiasFilterer.class);
+
+    //TODO: Promote or replace with VCFConstants call
+    public static final String ALT_F2R1 = "ALT_F2R1";
+    public static final String ALT_F1R2 = "ALT_F1R2";
+
+    public static final Double PRE_ADAPTER_METRIC_NOT_ARTIFACT_SCORE = 100.0;
+    public static final double BIASP = 0.96;
+
+    /** Adds the annotations that can be created for individual variants.  The annotations placed on the genotypes here will be used by the filter.
+     * @param variant the variant to prepare for the orientation bias filter. Never {@code null}
+     * @param relevantArtifactModesWithoutComplement the SNV artifact modes that we want to filter against.  Do not include the complement.  Never {@code null}
+     * @param preAdapterQScoreMap mapping from Artifact mode to the preAdapterQ score.  Never {@code null}
+     * @return updated VariantContext with new genotypes already populated.
+     */
+    public static VariantContext annotateVariantContextWithPreprocessingValues(final VariantContext variant, final SortedSet<ArtifactMode> relevantArtifactModesWithoutComplement, final Map<ArtifactMode, Double> preAdapterQScoreMap) {
+
+        Utils.nonNull(variant);
+        Utils.nonNull(relevantArtifactModesWithoutComplement);
+        Utils.nonNull(preAdapterQScoreMap);
+
+        final List<ArtifactMode> relevantArtifactModesComplement = OrientationBiasUtils.createReverseComplementArtifactModes(relevantArtifactModesWithoutComplement);
+
+        VariantContextBuilder variantBuilder = new VariantContextBuilder(variant);
+        GenotypesContext genotypesContext = variant.getGenotypes();
+
+        final List<Genotype> newGenotypes = new ArrayList<>();
+        for (Genotype genotype: genotypesContext.iterateInSampleNameOrder()) {
+            final GenotypeBuilder genotypeBuilder = new GenotypeBuilder(genotype);
+            genotypeBuilder.attribute(OrientationBiasFilterConstants.IS_ORIENTATION_BIAS_ARTIFACT_MODE, String.valueOf(false));
```

Comment:
I do use it in the end no matter what.  Not sure how to address this.  

No action.